src/utils/vsCodeGit.ts

```diff
@@ -0,0 +1,58 @@
+import * as vscode from "vscode"
+
+//https://github.com/microsoft/vscode/blob/main/extensions/git/src/api/git.d.ts
+interface GitRepository {
+	diff(staged: boolean): Promise<string>
+	readonly inputBox: { value: string }
+	readonly state: {
+		readonly indexChanges: GitFileChange[]
+	}
+}
+
+interface GitFileChange {
+	readonly status: number
+	readonly uri: vscode.Uri
+}
+
+export enum Status {
```

Comment:
can we use it with https://github.com/RooVetGit/Roo-Code/blob/main/src/utils/git.ts ?